### DISCUSSION TITLE
Keycloak and Outline : email required

### DISCUSSION THREAD

**Discussion by romstann** [outline/outline#6697] [Self Hosting]
Hi everyone, 

I'm trying to setup Outline since a week, and now I'm stuck on the authentification problem. 
I set up Keycloak, create a realm, a user, and when I try to login into outline, I get this error message : 

`{"error":"An email field was not returned in the profile parameter, but is required.","level":"error"`

Email is set up onto Keycloak for the user and here is my SCOPE : 
`OIDC_SCOPES="openid email"`
I've try already multiple things : 
`OIDC_SCOPES=openid email`
`OIDC_SCOPES=openid profile email`

I've found lots of issues regarding this email problem but no one get an answer with a solution that's work for me. 

In Keycloak, I found the session as active, authentification is working. 
I have activate user logs into Keycloak and LOGIN & CODE_TO_TOKEN are working fine. 
Into Client SCOPE : Email, Profile are on the "Default" status
Into Realm Configuration : Login with email is checked

both services are running on the same docker network. 

If you have any idea it could be very nice 🙏

**Comment by tommoor** [comment]
`OIDC_SCOPES` is the list of scopes that will be _requested_ from Keycloak. In this case although you are requesting the scope no email is being provided in the response from Keycloak so the solution will be in the configuration on that side.
